test/test_layouts.py

```diff
@@ -0,0 +1,14 @@
+from gradio.components import Button
+from gradio.layouts import Form, Row
+
+
+def test_form_min_width_guard():
+    """
+    Test that `Form.add_child` does not crash with a component whose `min_width` is `None`.
+    """
+    r = Row()
+    f = Form()
```

Comment:
this seems to test the internal implementation, using `.parent` and `.add_child`. How about this instead, which is how Blocks are naturally built?

```
with Row() as r:
  with Form() as f:
    b = Button() 
assert f.min_width == 0 
```

Reply:
@aliabid94 This doesn't trigger the original bug (for whichever reason) – I did mention I didn't have a great standalone repro :(

Reply:
That's really strange, can we figure out why the bug is not being triggered?

Reply:
I think it has to do with the `pseudo_parent` logic in `fill_expected_parents`, and truth be told my eyes glazed over trying to figure out what happens there. 😅

Reply:
I don't think this is a reasonable test as it uses an internal unsupported API. We have to reproduce the original problem with our supported API. Was the original bug discovered when using the API in this way as well?

Reply:
No, the original bug simply appears when trying to upgrade the `stable-diffusion-webui` repo to the latest version of Gradio. It doesn't use the internal APIs (but it does have some [special subclassed components](https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/modules/ui_components.py)).

Reply:
@akx as a general principle, we can't be making changes to the internals of the `gradio` library based on nonstandard usage. The fix or modification should be downstream library using `gradio`. If you can find an example where using `gradio` as intended produces this bug, then we're happy to fix, but in the meantime I think we should close this PR

Reply:
@abidlabs What constitutes "nonstandard usage"? 😁 As I see it, no non-public APIs are being used in the webui.

Reply:
We need a simple repro so we can test against regressions without testing implementation details.